### ISSUE TITLE
include single precision support in fftw

### ISSUE THREAD

**Issue by lschneiderbauer** [R-macos/recipes#61]
I would like to bring this [package](https://github.com/lschneiderbauer/fCWTr/) to CRAN, which depends on fftw including single precision support.

This works for CRAN Windows as well as Linux builds, but not for MacOS, since the default MacOS build does not include single precision support.

My suggestion is to modify the fftw recipe file to include a simple "Configure: --enable-float", enabling single precision by default. My understanding is that this would enable the automatic build of the above mentioned package for the MacOS platform when submitting to CRAN.

Would you be okay with a PR performing this simple change?

**Comment by s-u** [comment]
@lschneiderbauer Unfortunately that won't work - it would *remove* double precision support breaking everyone's code - you can only use one type in the `fftw` library. So one option would be to fix your package to use double precision as is the standard in R. The other would be to use a second, prefixed build of `fftw` (see [fftw docs](http://www.fftw.org/fftw3_doc/Precision.html)). If you can adjust your package to use the prefixed version of `fftw` (for you that would be `fftw3f` and the corresponding entry points) then I can add that to the recipes.

**Comment by lschneiderbauer** [comment]
Thank you for your help.

Unfortunately, the first option, although would probably be best, is practically out of my hands, since I am relying on upstream code that I do not control.

Regarding the second option: I think those pieces are already in place: The package's configure script should automatically pick up the library that contains the relevant single-precision functions (e.g. on Ubuntu, there is also a separate libfftw3f.so which gets picked up).

I see you already pushed a new recipe to master, thanks for that! <3
Can I already test those changes with the MacOS build service (https://mac.r-project.org/macbuilder/submit.html) or does it take a while until the changes land there?

**Comment by s-u** [comment]
@lschneiderbauer yes, please, test it, all binaries are up on all the CRAN machines

FWIW you check see if binaries are updated at https://mac.r-project.org/bin 
e.g. for arm64: https://mac.r-project.org/bin/darwin20/arm64/

**Comment by jeroen** [comment]
I have also updated the r-universe bundle, so users can test with the new libs there as well: https://github.com/r-universe-org/cranlibs/releases/tag/2025-01-05

**Comment by lschneiderbauer** [comment]
Great news,
the package compiles now on the MacOS builder: https://mac.r-project.org/macbuilder/results/1736159623-58e1880408414635/

(even though without OpenMP support, but that's another topic, optional, and not so important anyways)

Thanks!
